epochStart/metachain/systemSCs.go

```diff
@@ -471,7 +530,21 @@ func (s *systemSCProcessor) EpochConfirmed(epoch uint32) {
 
 	// only toggle on exact epoch. In future epochs the config should have already been synchronized from peers
 	s.flagHystNodesEnabled.Toggle(epoch == s.hystNodesEnableEpoch)
+
+	for _, maxNodesConfig := range s.maxNodesEnableConfig {
+		if epoch >= maxNodesConfig.EpochEnable {
+			// to cover also rollbacks, we always set the maxNodes and set the Enabled flag
+			s.flagChangeMaxNodesEnabled.Toggle(true)
```

Comment:
this will be called for all epochEnable, not just for the last one. If there is no way to process in parallel this and ProcessSystemSmartContract, there should be no problem, because the last set will be considered.